rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,255 @@
+            counterparty=CPT_AAVE_V2,
+        )
+
+    def _decode_deposit(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        user = hex_or_bytes_to_address(tx_log.data[:32])
+        on_behalf_of = hex_or_bytes_to_address(tx_log.data[64:96]) if hex_or_bytes_to_str(tx_log.data[64:96]) != '' else None  # noqa: E501
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            asset=token,
+        )
+        notes = f'Deposit {amount} {token.symbol} ({token.name}) from user {user} into AAVE v2'
+        if on_behalf_of is not None:
+            notes += f' on behalf of {on_behalf_of}'
+        for event in decoded_events:
+            if (
+                event.location_label == user and
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if event.event_type == HistoryEventType.SPEND:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.notes = notes
+                    event.counterparty = CPT_AAVE_V2
+
+                elif event.event_type == HistoryEventType.RECEIVE:
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} ({resolved_asset.name}) from AAVE v2 for {user}'  # noqa: E501
+                    event.counterparty = CPT_AAVE_V2
+
+    def _decode_withdrawal(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        to = hex_or_bytes_to_address(tx_log.topics[3])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Withdraw {amount} {token.symbol} ({token.name}) from AAVE v2 user {user}'
+        if to != user:
+            notes += f' to {to}'
+        for event in decoded_events:
+            if (
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if (
+                    event.location_label == to and
+                    event.event_type == HistoryEventType.RECEIVE
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.notes = notes
+                    event.counterparty = CPT_AAVE_V2
```

Comment:
The counterparty is set in both branches to the same value right? can be moved out of the if/else statement and avoid repetition